src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -15,38 +20,46 @@ def get_schema(self):
     def get_role(self):
         return self._conn.role
 
-    def get_repository_url_list(self, repo_name: str) -> SnowflakeCursor:
-        role = self.get_role()
+    def get_repository_row(self, repo_name: str) -> Dict:
+        if not is_valid_identifier(repo_name):
+            raise ValueError(
+                f"repo_name '{repo_name}' is not a valid Snowflake identifier"
+            )
         database = self.get_database()
         schema = self.get_schema()
 
-        registry_query = f"""
-            use role {role};
-            use database {database};
-            use schema {schema};
-            show image repositories like '{repo_name}';
-            """
-
-        return self._execute_query(registry_query)
-
-    def get_repository_url(self, repo_name):
-        database = self.get_database()
-        schema = self.get_schema()
+        # Unquoted identifiers are resolved as all upper case in Snowflake while quoted identifiers are case-sensitive
+        repo_name_raw = (
+            repo_name[1:-1]
+            if is_valid_quoted_identifier(repo_name)
+            else repo_name.upper()
+        )
+        repository_list_query = (
+            f"show image repositories like '{escape_like_pattern(repo_name_raw)}'"
+        )
+        result_set = self._execute_query(repository_list_query, cursor_class=DictCursor)
+        results = result_set.fetchall()
 
-        result_set = self.get_repository_url_list(repo_name=repo_name)
+        # because SHOW LIKE uses case-insensitive matching, results may return multiple rows
+        results = [r for r in results if r["name"] == repo_name_raw]
 
-        results = result_set.fetchall()
         if len(results) == 0:
-            raise ClickException(
+            raise ValueError(
                 f"Specified repository name {repo_name} not found in database {database} and schema {schema}"
             )
-        else:
```

Comment:
Copy SQL error format: Image Repository 'LADADA' does not exist or not authorized.